core/liquidity/v2/engine.go

```diff
@@ -225,11 +226,54 @@ func (e *Engine) SubmitLiquidityProvision(
 		UpdatedAt:        now,
 	}
 
-	e.provisions.Set(party, lp)
-	e.slaPerformance[party] = &slaPerformance{}
+	e.broker.Send(events.NewLiquidityProvisionEvent(ctx, provision))
 
-	e.broker.Send(events.NewLiquidityProvisionEvent(ctx, lp))
-	return nil
+	// add immediately during the opening auction
+	// otherwise schedule to be added at the beginning of new epoch
+	if e.auctionState.IsOpeningAuction() {
+		e.provisions.Set(party, provision)
+		e.slaPerformance[party] = &slaPerformance{}
+		return true, nil
+	}
+
+	provision.Status = types.LiquidityProvisionStatusPending
+	e.pendingProvisions[party] = provision
+	return false, nil
+}
+
+func (e *Engine) ApplyPendingProvisions(ctx context.Context, now time.Time) map[string]*types.LiquidityProvision {
+	updatedProvisionsPerParty := make(map[string]*types.LiquidityProvision, len(e.pendingProvisions))
+	for party, provision := range e.pendingProvisions {
```

Comment:
should probably sort this by party first so that the events are sent in deterministic order

Reply:
Defo